app/javascript/mastodon/service_worker/entry.js

```diff
@@ -54,8 +54,9 @@ self.addEventListener('fetch', function(event) {
 
       if (cached === undefined) {
         const fetched = await fetch(event.request);
+        const control = fetched.headers.get('Cache-Control');
```

Comment:
We can't read any headers for no-cors (opaque filtered) response.

https://fetch.spec.whatwg.org/#concept-filtered-response-opaque

> An opaque filtered response is a filtered response whose type is "opaque", url list is the empty list, status is 0, status message is the empty byte sequence, header list is empty, body is null, and trailer is empty.